src/hotspot/share/opto/superword.hpp

```diff
@@ -57,16 +57,281 @@
+  {
+    assert(message != nullptr, "must have message");
+    assert(_kind != Unchanged || split_size == 0, "unchanged task conditions");
+    assert(_kind != Rejected  || split_size == 0, "reject task conditions");
+    assert(_kind != Split     || split_size != 0, "split task conditions");
+  }
+
+public:
+  static SplitTask make_split(const uint split_size, const char* message) {
+    return SplitTask(Split, split_size, message);
+  }
+
+  static SplitTask make_unchanged() {
+    return SplitTask(Unchanged, 0, "unchanged");
+  }
+
+  static SplitTask make_rejected(const char* message) {
+    return SplitTask(Rejected, 0, message);
+  }
+
+  bool is_unchanged() const { return _kind == Unchanged; }
+  bool is_rejected() const { return _kind == Rejected; }
+  bool is_split() const { return _kind == Split; }
+  const char* message() const { return _message; }
+
+  uint split_size() const {
+    assert(is_split(), "only split tasks have split_size");
+    return _split_size;
+  }
+};
+
+class SplitStatus {
+private:
+  enum Kind {
+    // After split_pack, we have:                              first_pack   second_pack
+    Unchanged, // The pack is left in the pack, unchanged.     old_pack     nullptr
+    Rejected,  // The pack is removed from the packset.        nullptr      nullptr
+    Modified,  // The pack had some nodes removed.             old_pack     nullptr
+    Split,     // The pack was split into two packs.           pack1        pack2
+  };
+  Kind _kind;
+  Node_List* _first_pack;
+  Node_List* _second_pack;
+
+  SplitStatus(Kind kind, Node_List* first_pack, Node_List* second_pack) :
+    _kind(kind), _first_pack(first_pack), _second_pack(second_pack)
+  {
+    assert(_kind != Unchanged || (first_pack != nullptr && second_pack == nullptr), "unchanged status conditions");
+    assert(_kind != Rejected  || (first_pack == nullptr && second_pack == nullptr), "rejected status conditions");
+    assert(_kind != Modified  || (first_pack != nullptr && second_pack == nullptr), "modified status conditions");
+    assert(_kind != Split     || (first_pack != nullptr && second_pack != nullptr), "split status conditions");
+  }
+
+public:
+  static SplitStatus make_unchanged(Node_List* old_pack) {
+    return SplitStatus(Unchanged, old_pack, nullptr);
+  }
+
+  static SplitStatus make_rejected() {
+    return SplitStatus(Rejected, nullptr, nullptr);
+  }
+
+  static SplitStatus make_modified(Node_List* first_pack) {
+    return SplitStatus(Modified, first_pack, nullptr);
+  }
+
+  static SplitStatus make_split(Node_List* first_pack, Node_List* second_pack) {
+    return SplitStatus(Split, first_pack, second_pack);
+  }
+
+  bool is_unchanged() const { return _kind == Unchanged; }
+  Node_List* first_pack() const { return _first_pack; }
+  Node_List* second_pack() const { return _second_pack; }
+};
+
+class PackSet : public StackObj {
+private:
+  const VLoop& _vloop;
+  const VLoopBody& _body;
+
+  // The "packset" proper: an array of "packs"
```

Comment:
What do you mean by " The "packset" proper"?

Reply:
Changed comment to `// Set of all packs:`